pkg/cmd/util/clientcmd/factory_interface.go

```diff
@@ -0,0 +1,200 @@
+package clientcmd
+
+import (
+	"io"
+	"time"
+
+	"github.com/emicklei/go-restful/swagger"
+	"github.com/openshift/origin/pkg/client"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/validation"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// InterfaceFactory represents an interface for a Factory.
+type InterfaceFactory interface {
```

Comment:
FactoryInterface instead of InterfaceFactory? (it's not a factory for creating interfaces, it's an interface for a factory...I think?  though i'm not even totally clear on why it's being called a Factory)